commcare_connect/opportunity/utils/completed_work.py

```diff
@@ -0,0 +1,31 @@
+from commcare_connect.opportunity.models import CompletedWorkStatus
+
+
+def update_status_and_compute_payment(completed_works, opportunity, compute_payment=True):
+    """
+    Updates the status of completed works and optionally calculates total payment_accrued.
+    """
+    payment_accrued = 0
+    for completed_work in completed_works:
+        if completed_work.completed_count < 1:
+            continue
+
+        if opportunity.auto_approve_payments:
+            update_completed_work_status(completed_work)
+
+        if compute_payment:
+            approved_count = completed_work.approved_count
+            if approved_count > 0 and completed_work.status == CompletedWorkStatus.approved:
+                payment_accrued += approved_count * completed_work.payment_unit.amount
+    return payment_accrued
```

Comment:
It is a bit confusing that this function modifies some fields but returns others. I think it would be clearer if it just modified everything rather than forcing the caller to assign some values. It is also a bit unintuitive that it saves some models but not others. I would lean toward it just saving everything, unless there is a good reason even though that is generally not best practice. It could be controlled by a variable if that feels better

Reply:
@calellowitz Changed